secure/push-metrics/src/pusher.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::counters::Metrics;
+use libra_logger::{debug, error, info};
+use std::{env, sync::Arc, thread, thread::JoinHandle, time::Duration};
+
+const DEFAULT_PUSH_FREQUENCY_SECS: u64 = 15;
+
+/// MetricsPusher provides a function to push a list of Metrics to a configurable
+/// pushgateway endpoint.
+pub struct MetricsPusher {
+    metrics: Arc<dyn Metrics + Send + Sync>,
+}
+
+impl MetricsPusher {
+    pub fn new(metrics: Arc<dyn Metrics + Send + Sync>) -> Self {
+        Self { metrics }
+    }
+
+    fn run(self, push_metrics_endpoint: String, push_metrics_frequency_secs: u64) {
+        loop {
+            let mut data = "".to_string();
```

Comment:
String::new()?

Reply:
Done

Reply:
or look at using write!, I think that looks a little nicer, but this works.